cfn-resources/org-invitation/cmd/resource/resource.go

```diff
@@ -0,0 +1,241 @@
+	}
+	invitation, res, err := client.Organizations.InviteUser(context.Background(), *currentModel.OrgId, invitationReq)
+	if err != nil {
+		_, _ = log.Warnf("Create - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+	currentModel.Id = &invitation.ID
+
+	return Read(req, prevModel, currentModel)
+}
+
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("Read() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(ReadRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	invitation, res, err := client.Organizations.Invitation(context.Background(), *currentModel.OrgId, *currentModel.Id)
+	if err != nil {
+		_, _ = log.Warnf("Read - error: %+v", err)
+
+		// if invitation already accepted
+		if res.StatusCode == 404 {
+			if alreadyAccepted, _ := validateOrgInvitationAlreadyAccepted(context.Background(), client, *currentModel.Username, *currentModel.OrgId); alreadyAccepted {
+				return progressevents.GetFailedEventByResponse("invitation has been already accepted", res.Response), nil
+			}
+		}
+
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	readAtlasOrgInvitation(invitation, currentModel)
+	// Response
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Warnf("Update() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(UpdateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	invitationReq := &mongodbatlas.Invitation{
+		TeamIDs: currentModel.TeamIds,
+		Roles:   currentModel.Roles,
+	}
+
+	_, res, err := client.Organizations.UpdateInvitationByID(context.Background(), *currentModel.OrgId, *currentModel.Id, invitationReq)
+	if err != nil {
+		_, _ = log.Warnf("Update - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+	_, _ = log.Debugf("%s invitation updated", *currentModel.Id)
+
+	// Response
+	return Read(req, prevModel, currentModel)
```

Comment:
Instead of using READ api call, Can we leverage a common method just creating Model with necessary data.

Reply:
```suggestion
	// Response
	return handler.ProgressEvent{
		OperationStatus: handler.Success,
		ResourceModel:   readAtlasOrgInvitation(invitation, currentModel),
	}, nil


```